telemetry/telemetry.go

```diff
@@ -209,3 +235,17 @@ func getflag(name string) string {
 	})
 	return val
 }
+
+func signalHandler(reload chan<- int, wg *sync.WaitGroup, testSigChan <-chan os.Signal) {
+	defer wg.Done()
+	sigchannel := make(chan os.Signal, 1)
+	signal.Notify(sigchannel,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+	select {
```

Comment:
Can you explain sigchannel and testSigChan?

Reply:
Removed